src/components/QueryLoader/index.tsx

```diff
@@ -1,33 +1,46 @@
 import React, { useEffect, useState } from 'react';
 import { CircularProgress } from '@mui/material';
-import { useIsFetching } from '@tanstack/react-query';
+import { QueryKey, useIsFetching } from '@tanstack/react-query';
 
 export interface QueryLoaderProps {
-  queryKey: string;
+  queryKey?: string | QueryKey;
   children: React.ReactNode;
   loaderComponent?: React.ReactNode;
   /** minimum loading time in millis. useful for fake loading time or to avoid flickering  */
   atLeast?: number;
+  /** if provided the loading logic needs to managed outside this component via this boolean 'laoding' prop */
+  loading?: boolean;
 }
 
 /**
  * This simple component receive a queryKey identifier as prop and renders a loading spinner
  * until the relative fetch is in progress, otherwise it renders the children
  */
 const QueryLoader = (props: QueryLoaderProps) => {
-  const { queryKey, loaderComponent, children } = props;
-  const [atLeast, setLeast] = useState(0);
-  const isFetching = useIsFetching({ queryKey: [queryKey] });
+  const { queryKey, loaderComponent, children, loading } = props;
+  const [atLeast, setAtLeast] = useState(props.atLeast || 0);
+  const isFetching = useIsFetching({
+    queryKey: typeof queryKey === 'string' ? [queryKey] : queryKey
+  });
 
   useEffect(() => {
     if (props.atLeast && isFetching) {
-      setLeast(props.atLeast);
-      setTimeout(() => setLeast(0), props.atLeast);
+      setTimeout(() => setAtLeast(0), props.atLeast);
     }
   }, [isFetching, props.atLeast]);
 
   const loader = loaderComponent || <CircularProgress />;
-  return <>{isFetching + atLeast ? loader : children}</>;
+  return (
+    <>
+      {loading !== undefined
```

Comment:
We could add a local variable like:
```js
const isLoading = loading?? isFetching || atLeast > 0
```
and change this ternary with
```js
 isLoading? loader:children
 ```

Reply:
What you are suggesting to me implies a different logic. 
if I apply the suggestions you give me the functioning is no longer as desired